src/tree.jl

```diff
@@ -4,6 +4,60 @@ mutable struct PruneData
     prune_threshold::Float64
 end
 
+struct BinManager
+    lowest_ub::Float64
+    num_levels::Int
+    num_of_bins_per_level::Vector{Int}
+    bin_levels_intervals::Vector{Dict{Symbol, Float64}}
+    bin_levels_nodes::Vector{Dict{Int, Dict{Symbol, Union{Tuple{Int, Int}, Float64}}}}
+    bin_levels::Vector{Dict{Symbol, Dict{Tuple{Int, Int}, Union{Float64, Int}}}}
```

Comment:
These vectors of dictionaries (nested with further dictionaries) are really expensive in terms of performance. Also, the union typing results in some type instabilities.

Provided that a lot of the keys for these dictionaries are always the same (e.g. bin_value, bin_count, bin_error), can you find any way to flatten everything out into vectors, use fewer dictionaries, or just use structs/named tuples?

Reply:
Yeah. I wasn't a big fan of the dictionary implementation when I was doing it. Reworking as vectors wouldn't be too bad since the number of bins stays the same. I can mark this as WIP until I make these changes if desired.

Reply:
I changed the implementation with [3762018](https://github.com/JuliaPOMDP/NativeSARSOP.jl/pull/18/commits/37620188f4c5054d8a3fd1b7bd1a3b0b5bbc6430). In the process, I also found a bug with the dictionaries. With the binning, we are back down to the normal level of allocations vs the original implementation.

I updated the performance comparisons and added results for both `delta=0.1` and `delta=0.0001`.